### PR TITLE
[bundle side]support test explorer

### PR DESCRIPTION
**Pull request by ansyral**
Signed-off-by: xuzho <xuzho@microsoft.com>

### CODE REVIEW

**Comment by ansyral** [comment]
#45

**Comment by akaroml** [comment]
+ @testforstephen please check this one. Jinbo made changes to the search mechanism which boost performance.

File: com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/JUnitTestSearcher.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+					IType type = member.getElementType() == IJavaElement.TYPE ? (IType) member
+							: member.getDeclaringType();
+					testClasses.add(type);
+				}
+			}
+		};
+
+		try {
+		    IJavaSearchScope scope = createSearchScope();
+			new SearchEngine().search(
```
Comment (testforstephen) [inline com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/JUnitTestSearcher.java]: indentation is not right.

File: com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/JUnitTestSearcher.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import org.eclipse.jdt.core.search.SearchRequestor;
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+import org.eclipse.jdt.ls.core.internal.ResourceUtils;
+import org.eclipse.lsp4j.Range;
+
+import com.microsoft.java.test.plugin.internal.testsuit.TestLevel;
+import com.microsoft.java.test.plugin.internal.testsuit.TestSuite;
+
+public class JUnitTestSearcher {
+	private final String JUNIT_TEST_ANNOTATION = "org.junit.Test";
```
Comment (testforstephen) [inline com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/JUnitTestSearcher.java]: The indentation should be 4 spaces.

File: com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/JUnitTestSearcher.java (hunk 1 above)
Comment (akaroml) [reply com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/JUnitTestSearcher.java]: +1

Comment (ansyral) [reply com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/JUnitTestSearcher.java]: ok, i mixed tab and whitespace

File: com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/JUnitTestSearcher.java (hunk 2 above)
Comment (ansyral) [reply com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/JUnitTestSearcher.java]: it is four, not sure why github rendered it wrong

**Comment by ansyral** [comment]
let me change the indentation from tab to space in another pr

**Pull request merged**